python/tvm/relay/frontend/sklearn.py

```diff
@@ -0,0 +1,154 @@
+
+
+def _SimpleImputer(op, inexpr, dshape, dtype, columns=None):
+    boolean_mask = _op.isnan(inexpr)
+    fill_col = _op.const(np.array(op.statistics_, dtype=dtype))
+    input_shape = _op.shape_of(inexpr)
+    reps = _op.take(input_shape, _op.const([0]))
+    reps = _op.concatenate([reps, _op.const([1])], axis=0)
+
+    fill_val = _op.tile(fill_col, reps=reps)
+    indices =_op.const(np.arange(len(op.statistics_)))
+    fill_val = _op.take(fill_val, indices=indices, axis=1)
+
+    ret = _op.where(boolean_mask,
+                    fill_val,
+                    inexpr)
+    
+    return ret
+
+def _RobustImputer(op, inexpr, dshape, dtype, columns=None):
+    if columns: 
+        column_indices = _op.const(columns)
+        inexpr = _op.take(inexpr, indices=column_indices, axis=1)
+
+    if not op.mask_function:
+        inf_mask = _op.isinf(inexpr)
+        nan_val = _op.full_like(inexpr, _op.const(np.array(np.nan, dtype=dtype)))
+        inexpr = _op.where(inf_mask, nan_val, inexpr) 
+    ret = _SimpleImputer(op.simple_imputer_, inexpr, dshape, dtype, columns)
+
+    return ret 
+    
+def _ThresholdOneHotEncoder(op, inexpr, dshape, dtype, columns=None):
+    if columns: 
+        column_indices = _op.const(columns)
+        inexpr = _op.take(inexpr, indices=column_indices, axis=1)
+
+    num_cat = len(op.categories_)
+    cols = _op.split(inexpr, num_cat, axis=1)
+
+    out = [] 
+    for i in range(num_cat):
+        category = op.categories_[i]
+        cat_tensor = _op.const(np.array(category, dtype=dtype))
+        tiled_col = _op.tile(cols[i], (1, len(category)))
+        one_hot_mask = _op.equal(tiled_col, cat_tensor)
+        one_hot = _op.cast(one_hot_mask, dtype)
+        out.append(one_hot)
+
+    ret = _op.concatenate(out, axis=1) 
+    return ret
+
+def _RobustStandardScaler(op, inexpr, dshape, dtype, columns=None):
+    scaler = op.scaler_
+    ret = _op.subtract(inexpr, _op.const(np.array(scaler.mean_, dtype), dtype))
+    ret = _op.divide(ret, _op.const(np.array(scaler.scale_, dtype), dtype))
+    return ret
+
+def _ColumnTransformer(op, inexpr, dshape, dtype, columns=None):
+    out = []
+    for _, pipe, cols in op.transformers_:
+        mod = pipe.steps[0][1]
+        out.append(sklearn_op_to_relay(mod, inexpr, dshape, dtype, cols))
+    
+    return _op.concatenate(out, axis=1)
+
+_convert_map = {
+    'ColumnTransformer':_ColumnTransformer,
+    'SimpleImputer': _SimpleImputer,
+    'RobustImputer': _RobustImputer,
+    'RobustStandardScaler': _RobustStandardScaler,
+    'ThresholdOneHotEncoder': _ThresholdOneHotEncoder
+}
+
+def sklearn_op_to_relay(op, inexpr, dshape, dtype, columns=None):
+    classname = type(op).__name__
+    return _convert_map[classname](op, inexpr, dshape, dtype, columns)
+
+def from_sklearn(model,
+              shape=None,
+              dtype="float32",
+              columns=None):
+
+    try:
+        import sklearn
+    except ImportError:
+        pass
```

Comment:
Fail here?

Reply:
Added "Raising an Error when Sklearn is not installed". Please check